app/auth/gitlab_auth.py

```diff
@@ -40,198 +36,80 @@
-        "response_type": "code",
-        "scope": SCOPE,
-        "redirect_uri": current_app.config["HOST_NAME"] + url_for("gitlab_auth.token"),
-        "state": state,
-    }
-    auth_req = gitlab_oic_client.construct_AuthorizationRequest(request_args=args)
-    login_url = auth_req.request(gitlab_oic_client.authorization_endpoint)
-    response = current_app.make_response(redirect(login_url))
-    return response
 
 
 @blueprint.route("/token")
 def token():
-    from .. import store
-
-    gitlab_oic_client = current_app.config.get("GITLAB_OIC_CLIENT")
-
-    # This is more about parsing the request data than any response data....
-    authorization_parameters = gitlab_oic_client.parse_response(
-        AuthorizationResponse,
-        info=request.query_string.decode("utf-8"),
-        sformat="urlencoded",
-    )
-
-    if session["gitlab_state"] != authorization_parameters["state"]:
-        return "Something went wrong while trying to log you in."
-
-    token_response = gitlab_oic_client.do_access_token_request(
-        scope=SCOPE,
-        state=authorization_parameters["state"],
-        request_args={
-            "code": authorization_parameters["code"],
-            "redirect_uri": current_app.config["HOST_NAME"]
-            + url_for("gitlab_auth.token"),
-        },
-    )
-
-    a = jwt.decode(session["token"], verify=False)
-    store.put(
-        get_key_for_user(a, "gl_access_token"), token_response["access_token"].encode()
-    )
-    store.put(
-        get_key_for_user(a, "gl_refresh_token"),
-        token_response["refresh_token"].encode(),
-    )
-    store.put(
-        get_key_for_user(a, "gl_id_token"),
-        json.dumps(token_response["id_token"].to_dict()).encode(),
-    )
-
-    response = current_app.make_response(
-        redirect(current_app.config["HOST_NAME"] + url_for("web_auth.login_next"))
-    )
-
-    return response
-
-
-def get_gitlab_refresh_token(access_token):
-    from .. import store
-
-    gitlab_oic_client = current_app.config.get("GITLAB_OIC_CLIENT")
-
-    access_token = jwt.decode(
-        access_token,
-        current_app.config["OIDC_PUBLIC_KEY"],
-        algorithms=JWT_ALGORITHM,
-        audience=current_app.config["OIDC_CLIENT_ID"],
-    )
-    to = Token(
-        resp={
-            "refresh_token": store.get(
-                get_key_for_user(access_token, "gl_refresh_token")
-            )
-        }
-    )
-    refresh_token_response = gitlab_oic_client.do_access_token_refresh(token=to)
-    if "access_token" in refresh_token_response:
-        store.put(
-            get_key_for_user(access_token, "gl_access_token"),
-            refresh_token_response["access_token"].encode(),
-        )
-        store.put(
-            get_key_for_user(access_token, "gl_refresh_token"),
-            refresh_token_response["refresh_token"].encode(),
-        )
-    return refresh_token_response.get("access_token")
+    return handle_token_request(request, GL_SUFFIX)[0]
```

Comment:
I find a usage of destructuring here more readable and more clean in stating the intent.

```
response, _ = handle_token_request(request, GL_SUFFIX)
return response
```

Reply:
done